pdp/handlers.go

```diff
@@ -575,22 +591,42 @@ func (p *PDPService) handleAddRootToProofSet(w http.ResponseWriter, r *http.Requ
 		Subroots []SubrootEntry `json:"subroots"`
 	}
 
-	var req []AddRootRequest // array because we can add multiple roots in one request
-	err = json.NewDecoder(r.Body).Decode(&req)
+	// AddRootsPayload defines the structure for the entire add roots request payload
+	type AddRootsPayload struct {
+		Roots     []AddRootRequest `json:"roots"`
+		ExtraData *string          `json:"extraData,omitempty"`
+	}
+
+	var payload AddRootsPayload
+	err = json.NewDecoder(r.Body).Decode(&payload)
 	if err != nil {
 		http.Error(w, "Invalid request body: "+err.Error(), http.StatusBadRequest)
 		return
 	}
 	defer r.Body.Close()
 
-	if len(req) == 0 {
+	if len(payload.Roots) == 0 {
 		http.Error(w, "At least one root must be provided", http.StatusBadRequest)
 		return
 	}
 
+	var extraDataBytes []byte
+	if payload.ExtraData != nil {
+		extraDataHexStr := *payload.ExtraData
+		decodedBytes, err := hex.DecodeString(strings.TrimPrefix(extraDataHexStr, "0x"))
+		if err != nil {
+			log.Errorf("Failed to decode hex extraData: %v", err)
+			http.Error(w, "Invalid extraData format (must be hex encoded)", http.StatusBadRequest)
+			return
+		}
+		extraDataBytes = decodedBytes
+	} else {
+		extraDataBytes = []byte{} // Ensure it's an empty byte slice if not provided
```

Comment:
nit: just initialize extraDataBytes := []byte{} on line 613.  Then we can avoid the else statement